File: Programmers/카펫.py
brown=24
yellow=24

def solution(brown, yellow):
    answer = []
    weight=brown+yellow

    for i in range(brown,2,-1):
        for j in range(brown,2,-1):
            if yellow == (i-2)*(j-2):
                if weight==i*j:
                    answer.append(i)
                    answer.append(j)
                    return answer


    return answer

print(solution(brown,yellow))

File: Programmers/카펫.cpp
#include <bits/stdc++.h>


using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer;
    int weight=brown+yellow;

    for(int i=brown;i>2;i--){
        for(int j=brown;j>2;j--){
            if (yellow==(i-2)*(j-2)){
                if(weight==i*j){
                    answer.push_back(i);
                    answer.push_back(j);
                    return answer;
                }
            }
        }
    }

}
